services/src/test/java/io/druid/cli/validate/DruidJsonValidatorTest.java

```diff
@@ -121,6 +142,76 @@ public void write(String str)
     Assert.assertEquals(expected, writer.toString());
   }
 
+  @Test
+  public void testTaskValidator() throws Exception
+  {
+    final ObjectMapper jsonMapper = new DefaultObjectMapper();
+    for (final Module jacksonModule : new FirehoseModule().getJacksonModules()) {
+      jsonMapper.registerModule(jacksonModule);
+    }
+
+    final RealtimeIndexTask task = new RealtimeIndexTask(
+        null,
+        new TaskResource("rofl", 2),
+        new FireDepartment(
+            new DataSchema(
+                "foo",
+                null,
+                new AggregatorFactory[0],
+                new UniformGranularitySpec(Granularity.HOUR, QueryGranularity.NONE, null),
+                jsonMapper
+            ),
+            new RealtimeIOConfig(
+                new LocalFirehoseFactory(new File("lol"), "rofl", null), new PlumberSchool()
+            {
+              @Override
+              public Plumber findPlumber(
+                  DataSchema schema, RealtimeTuningConfig config, FireDepartmentMetrics metrics
+              )
+              {
+                return null;
+              }
+            },
+                null
+            ),
+
+            new RealtimeTuningConfig(
+                1,
+                new Period("PT10M"),
+                null,
+                null,
+                null,
+                null,
+                1,
+                new NoneShardSpec(),
+                new IndexSpec(),
+                null,
+                0,
+                0,
+                true,
+                null
+            )
+        ),
+        null
+    );
+
+    File tmp = File.createTempFile("test", null);
```

Comment:
we should cleanup after ourselves and remove the temp file. This test suite already has a temp folder, not sure if we should reuse that or not, I don't really understand what it's used for.